nifi-ngsi-bundle/nifi-ngsi-processors/src/main/java/org/apache/nifi/processors/ngsi/ngsi/utils/NGSIUtils.java

```diff
@@ -79,47 +79,105 @@ public NGSIEvent getEventFromFlowFile(FlowFile flowFile, final ProcessSession se
+                        String key = keys.next();
+                        if (!IGNORED_KEYS_ON_ENTITES.contains(key)) {
+                            Object object = entity.get(key);
+                            if (object instanceof JSONObject) {
+                                // it is an attribute with one instance only
+                                JSONObject value = entity.getJSONObject(key);
                                 AttributesLD attributesLD = parseNgsiLdAttribute(key, value);
                                 attributes.add(attributesLD);
+                            } else if (object instanceof JSONArray) {
+                                // it is a multi-attribute (see section 4.5.5 in NGSI-LD specification)
+                                JSONArray values = entity.getJSONArray(key);
+                                for (int j = 0; j < values.length(); j++) {
+                                    JSONObject value = values.getJSONObject(j);
+                                    AttributesLD attributesLD = parseNgsiLdAttribute(key, value);
+                                    attributes.add(attributesLD);
+                                }
+                            } else {
+                                logger.warn("Attribute {} has unexpected value type: {}", key, object.getClass());
                             }
-                        } else {
-                            logger.warn("Attribute {} has unexpected value type: {}", key, object.getClass());
                         }
                     }
+                    entities.add(new Entity(entityId,entityType,attributes,true));
                 }
-                entities.add(new Entity(entityId,entityType,attributes,true));
+            } else {
+                JSONArray content = new JSONArray(flowFileContent);
+                logger.debug("Received an NGSI-LD temporal data");
+                entities = parseNgsiLdEntities(content);
             }
             event = new NGSIEvent(creationTime,fiwareService,entities);
         }
         return event;
     }
 
+    public ArrayList<Entity> parseNgsiLdEntities(JSONArray content) {
+        ArrayList<Entity> entities = new ArrayList<>();
+        String entityType;
+        String entityId;
+        for (int i = 0; i < content.length(); i++) {
+            JSONObject temporalEntity = content.getJSONObject(i);
+            entityId = temporalEntity.getString("id");
+            entityType = temporalEntity.getString("type");
+            logger.debug("Dealing with entity {} of type {}", entityId, entityType);
+            ArrayList<AttributesLD> attributes  = new ArrayList<>();
+            Iterator<String> keys = temporalEntity.keys();
+            while (keys.hasNext()) {
+                String key = keys.next();
+                if (!IGNORED_KEYS_ON_ENTITES.contains(key)) {
+                    Object object = temporalEntity.get(key);
+                    if (object instanceof JSONObject) {
```

Comment:
this function is only used to parse temporal entities? If so, you can't have a JSON object?

Reply:
Yes I can remove this part